### PR TITLE
Add AN-C PCB

### PR DESCRIPTION
**Pull request by awkannan**
## Description

The AN-C PCB is a 60% PCB by CannonKeys.

It will be in stock soon.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/an_c/an_c.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "quantum.h"
+
+#define KNO KC_NO
+
+#define LAYOUT_ansi( \
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/an_c.h]:
```suggestion
#define LAYOUT_60_ansi( \
```

File: keyboards/cannonkeys/an_c/an_c.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  K40, K41,  K42,              K45,            K49, K4A, K4B,           K4E \
+) { \
+  {  K00,  K01,  K02,  K03,  K04,    K05,    K06,    K07,    K08,    K09,  K0A,  K0B,  K0C,  K0D,  KNO}, \
+  {  K10,  K11,  K12,  K13,  K14,    K15,    K16,    K17,    K18,    K19,  K1A,  K1B,  K1C,  KNO,  K1E   }, \
+  {  K20,  K21,  K22,  K23,  K24,    K25,    K26,    K27,    K28,    K29,  K2A,  K2B,  KNO,  KNO,  K2E  }, \
+  {  K30,  K31,  K32,  K33,  K34,    K35,    K36,    K37,    K38,    K39,  K3A,  K3B,  KNO,  KNO,  KNO  }, \
+  {  K40,  K41,  K42,  KNO,  KNO,    K45,    KNO,    KNO,    KNO,    K49,  K4A,  K4B,  KNO,  KNO,  K4E  }  \
+}
+
+#define LAYOUT_tsangan( \
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/an_c.h]:
```suggestion
#define LAYOUT_60_tsangan_hhkb( \
```

File: keyboards/cannonkeys/an_c/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// entirely and just use numbers.
+#define _BASE 0
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_ansi(
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/keymaps/default/keymap.c]:
```suggestion
  [_BASE] = LAYOUT_60_ansi(
```

File: keyboards/cannonkeys/an_c/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS, \
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT,  \
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                    KC_RSFT, \
+    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                          KC_RALT, KC_RGUI, MO(_FN1), KC_RCTL
+  ),
+
+  [_FN1] = LAYOUT_ansi(
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/keymaps/default/keymap.c]:
```suggestion
  [_FN1] = LAYOUT_60_ansi(
```

File: keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// entirely and just use numbers.
+#define _BASE 0
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_tsangan(
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c]:
```suggestion
  [_BASE] = LAYOUT_60_tsangan_hhkb(
```

File: keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSLS, KC_DEL, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSPC, \
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT,  \
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                    KC_RSFT, MO(_FN1),\
+    KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                          KC_RALT, KC_RGUI, KC_RCTL
+  ),
+
+  [_FN1] = LAYOUT_tsangan(
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c]:
```suggestion
  [_FN1] = LAYOUT_60_tsangan_hhkb(
```

File: keyboards/cannonkeys/an_c/rules.mk [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+# BACKLIGHT_ENABLE = yes # This is broken on 072 for some reason
+RGBLIGHT_ENABLE = yes
+
+# RAW_ENABLE = yes
+# DYNAMIC_KEYMAP_ENABLE = yes
+
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/rules.mk]:
```suggestion

LAYOUTS = 60_ansi 60_tsangan_hhkb
```

File: keyboards/cannonkeys/instant60/readme.md [hunk 8]
```diff
@@ -5,6 +5,8 @@ Instant60 Keyboard
 Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan1)
```
Comment (noroadsleft) [inline keyboards/cannonkeys/instant60/readme.md]:
```suggestion
Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
```

File: keyboards/cannonkeys/an_c/readme.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# AN-C
+
+AN-C Keyboard
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan1)
```
Comment (noroadsleft) [inline keyboards/cannonkeys/an_c/readme.md]: You seem to have deleted the `awkannan1` account from GitHub, so the existing link comes back Error 404.

```suggestion
Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
```

**Comment by yanfali** [comment]
Why does this say AN-60 and also include ortho75?

**Comment by awkannan** [comment]
I added both. I did the AN-C first, but it wasn't merged before Ortho75 was done so I added it to this one too

File: keyboards/cannonkeys/ortho75/readme.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# Ortho 74
```
Comment (zvecr) [inline keyboards/cannonkeys/ortho75/readme.md]:
```suggestion
# Ortho 75
```

File: keyboards/cannonkeys/an_c/info.json [hunk 11]
```diff
@@ -0,0 +1,15 @@
+{
+    "keyboard_name": "AN-C",
+    "url": "https://cannonkeys.com",
+    "maintainer": "awkannan",
+    "width": 15,
+    "height": 5,
+    "layouts": {
+        "LAYOUT_ansi": {
```
Comment (zvecr) [inline keyboards/cannonkeys/an_c/info.json]:
```suggestion
        "LAYOUT_60_ansi": {
```

File: keyboards/cannonkeys/an_c/info.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        },
+        "LAYOUT_tsangan": {
```
Comment (zvecr) [inline keyboards/cannonkeys/an_c/info.json]:
```suggestion
        "LAYOUT_60_tsangan_hhkb": {
```

**Comment by yanfali** [comment]
> I added both. I did the AN-C first, but it wasn't merged before Ortho75 was done so I added it to this one too

Generally tailgating one PR on top of another is a bit messy. Can you just separate the two please?

**Comment by yanfali** [comment]
Ortho75 has moved to PR 6177 please try and review that one too.

File: keyboards/cannonkeys/an_c/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+#define _BASE 0
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_60_ansi(
+    KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, \
```
Comment (fauxpark) [inline keyboards/cannonkeys/an_c/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater and less "hack-y".

File: keyboards/cannonkeys/an_c/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _BASE 0
+#define _FN1 1
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
```
Comment (fauxpark) [inline keyboards/cannonkeys/an_c/keymaps/default/keymap.c]: This keycode and the one in the tsangan keymap are not used, so these blocks can be removed.

**Comment by awkannan** [comment]
@fauxpark fixed

File: keyboards/cannonkeys/an_c/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/cannonkeys/an_c/keymaps/default/keymap.c]: Whoops, these can be removed too.

```suggestion
```

File: keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (fauxpark) [inline keyboards/cannonkeys/an_c/keymaps/tsangan/keymap.c]:
```suggestion
```

**Comment by drashna** [comment]
Most everything looks good.  Though, you may want to update the info.json file for the AN-C.  Right now, it looks like it won't render correctly, since it's referencing the layout macros incorrectly. 

I can merge as is, but I would prefer to see this fixed, before doing so

**Comment by drashna** [comment]
It looks like the rules.mk for this board don't include a `DFU_ARGS`, which is used for flashing.

Was this intentional, or an oversight?

**Comment by yanfali** [comment]
The actual info.json is correct, just update the layout names and it should be good to go

**Comment by awkannan** [comment]
I added all those things. @drashna

**Comment by noroadsleft** [comment]
@zvecr's [review](https://github.com/qmk/qmk_firmware/pull/6157#pullrequestreview-253110408) still has outstanding comments regarding `info.json` (which are things I missed on my review), but everything else here looks good to go.

**Comment by awkannan** [comment]
@noroadsleft Forgot to push! All good now

**Pull request merged**